packages/terra-framework-docs/src/terra-dev-site/doc/worklist-data-grid/Examples.2/RowSelection.jsx

```diff
@@ -104,6 +226,24 @@ const RowSelection = () => {
     }
   };
 
+  const onRowSelect = (rowIdsToSelect, rowIdsToUnselect) => {
+    if (rowIdsToUnselect) {
+      rows.forEach((r, index) => { if (rowIdsToUnselect.indexOf(r.id) >= 0) { rows[index].isSelected = false; } });
+    }
+
+    if (rowIdsToSelect) {
+      rows.forEach((r, index) => { if (rowIdsToSelect.indexOf(r.id) >= 0) { rows[index].isSelected = true; } });
+      setSelectedRows(rowIdsToSelect);
+    }
+  };
+
+  const enableRowSelection = () => {
+    if (!rowSelectionModeRef.current.checked) {
+      rowSelectionModeRef.current.checked = true;
+    }
+    setHasSelectableRows(true);
+  };
```

Comment:
The functions in your example should probably leverage the useCallback hook.

Reply:
Could you elaborate why and it's need?

Reply:
Discussed offline. Updated